torch/distributed/utils.py

```diff
@@ -0,0 +1,56 @@
+import torch
+
+def _parse_remote_device(remote_device: str):
+    r"""
+    Parses the remote device.
+
+    Args:
+        remote_device (str): Device on the destination worker where we'd like to place this module.
+            The format should be one of the following:
+
+                1. "<workername>/<device>", where the device field can be parsed as torch.device type.
+                   E.g., "trainer0/cpu", "trainer0", "ps0/cuda:0".
+                   In addition, the device field can be optional and the default value is "cpu".
+                2. "rank:<rank>/<device>", where <rank> is the rank of the
+                   process and device can be parsed as torch.device type.
+                   E.g., "rank:0/cpu", "rank:0", "rank:0/cuda:0"
+
+    Returns:
+        A workername/rank and a device.
+    """
+
+    PARSE_ERROR = (
+        f"Could not parse remote_device: {remote_device}. The valid format is "
+        "'<workername>/<device>' or 'rank:<rank>/<device>'"
+    )
+
+    fields = remote_device.split("/")
+    if len(fields) == 2:
```

Comment:
Looks like this method only checks whether the format is correct. Do we also need to check whether the ranks are valid before it is used by, e.g., DevicePlacementSpec?

Reply:
The specification itself might be tied to different runtime environments. For example, you can store this specification with a checkpoint and then load it during checkpoint load time. As a result, I thought its better not to validate runtime environment here. For example, we don't verify GPU indexes either.

Reply:
I see. That means, we consider the sharding spec as counterparts of Device/RemoteDevice, and the check will be enforced at the loading time?

Reply:
Right that is what I had in mind.

Reply:
Agree. This syntax-only non-semantics check is consistent with the remote device validation in RPC framework.